### ISSUE TITLE
Clippy suggests wrongly when `unwrap_or` takes an empty slice

### ISSUE THREAD

**Issue by profetia** [rust-lang/rust-clippy#15752]
### Summary

Clippy suggests wrongly when `unwrap_or` takes an empty slice, as after applying the fix, the default got type infered to &[T; 0] instead of a slice

### Reproducer

I tried this code:

```rust
#[warn(clippy::map_unwrap_or)]
fn bar() {
    struct Foo<'a>(&'a [u32]);

    let x = Some(Foo(&[1, 2, 3]));
    x.map_or(&[], |y| y.0);
}
```

I expected to see this happen:

suggests some like `&[][..]` or `<&[_]>::default` or may be no fix at all

Instead, this happened:

```
warning: called `map(<f>).unwrap_or(<a>)` on an `Option` value
 --> src/main.rs:8:5
  |
8 |     x.map(|y| y.0).unwrap_or(&[]);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_unwrap_or
note: the lint level is defined here
 --> src/main.rs:3:8
  |
3 | #[warn(clippy::map_unwrap_or)]
  |        ^^^^^^^^^^^^^^^^^^^^^
help: use `map_or(<a>, <f>)` instead
  |
8 -     x.map(|y| y.0).unwrap_or(&[]);
8 +     x.map_or(&[], |y| y.0);
  |
```

### Version

```text
rustc 1.92.0-nightly (4645a7988 2025-09-17)
binary: rustc
commit-hash: 4645a7988177c286f61609cc667ecae4c571a2e8
commit-date: 2025-09-17
host: x86_64-unknown-linux-gnu
release: 1.92.0-nightly
LLVM version: 21.1.1
```
